### ISSUE TITLE
In `from`, allow multiple `group` and `where` clauses in any order, optionally followed by `yield`

### ISSUE THREAD

**Issue by julianhyde** [julianhyde/morel#21]
Currently if a `from` expression has a `group` we do not allow `yield`.

As discussed in #20, we could allow `yield` by consisting the group keys and compute expressions to be variables. The variables defined in `in` sub-clauses of `from` are no longer visible.

For example,
```
from e in emps
  group e.deptno, e.job compute sum of e.sal as sumSal
```
defines variables `deptno`, `job`, `sumSal` and obscures `e`. Per the usual rules of `yield`, the query is implicitly followed by
```
yield {deptno, job, sumSal}
```
Again, per the usual rules of `yield`, if there is exactly one variable the record is elided. Thus
```
from e in emps
  group e.deptno
```
is equivalent to
```
from e in emps
  group e.deptno
  yield deptno
```
and
```
from e in emps
  group compute sum of e.sal as sumSal
```
is equivalent to
```
from e in emps
  group compute sum of e.sal as sumSal
  yield sumSal
```

Now we have the variables, we can write explicit `yield` clauses. For example,
```
from e in emps
  group e.deptno, e.job compute sum of e.sal as sumSal
  yield {deptno, job = substring (job, 0, 2)}
```

A reference to `e` will resolve to a variable in an enclosing scope:
```
let
  val e = 2.71828
in
  from e in emps
    group e.deptno
    yield {deptno, e}
end
```
will return something like `[{deptno = 10, e = 2.71828}, {deptno = 20, e = 2.71828}]`.

**Comment by julianhyde** [comment]
[Erik Meijer pointed out](https://twitter.com/headinthebox/status/1238626966367907840) that VB LINQ is compositional:

> In VB you can always add a new clause at the end of a query.
>
> In C# it has a fixed structure `from ... select`. Because of nesting it is of course compositional as well, but often not as elegant as in VB.

I realized that we could do something similar in Morel. After  `from` you can have as many `where` and `group` (and soon `order`) clauses as you like.  And you can optionally follow with `yield`. (If you omit `yield`, you get the default yield expression, which is a record whose fields are the current iteration variables, or a scalar if there is only one iteration variable.)

You can't have anything after `yield` because it doesn't assign to a variable. `where` and `order` propagate their incoming variables. `group` creates new variables (for the grouping columns and aggregate functions) and obscures the incoming variables.

(I think our `group` syntax is superior to VB's. We assign to multiple named key fields, rather than creating a single field that is always called `GROUP`.)

For example, the following uses two `where` clauses, the second being the equivalent of a SQL `HAVING`. Return (department, job) combinations with 5 or more employees, ordered within each department by the job with the most employees:
```
from e in emps, d in depts
  where e.deptno = d.deptno
  group d.dname, e.job compute count as c
  where c >= 5
  order dname, c desc
  yield {dname, job}
```

`order` is not implemented yet, but soon will be (see #20).

**Comment by julianhyde** [comment]
Fixed in a6d3209.
